tensorflow/lite/micro/examples/recipes/recipes_test.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+#
+import numpy as np
+import tensorflow as tf
+
+from tensorflow.python.framework import test_util
+from tensorflow.python.platform import test
+from tflite_micro.tensorflow.lite.micro.examples.recipes import resource_variables_lib
+from tflite_micro.tensorflow.lite.micro.python.interpreter.src import tflm_runtime
+
+
+class RecipesTest(test_util.TensorFlowTestCase):
+
+  def _use_tflite_interpreter(self, filename, input_data_list):
+    tflite_interpreter = tf.lite.Interpreter(
+        model_path=filename,
+        # TODO(b/): Latest TF pypi release package does not include registerer
+        # change needed to use BUILTIN_REF(CALL_ONCE op). Update once included.
+        experimental_op_resolver_type=tf.lite.experimental.OpResolverType.AUTO,
+    )
+    tflite_interpreter.allocate_tensors()
+    output_list = []
+    for input_dataset_index in range(3):
+      for idx, input_details in enumerate(
+          tflite_interpreter.get_input_details()
+      ):
+        input_tensor_idx = input_details["index"]
+        input_data = np.reshape(
+            input_data_list[input_dataset_index][idx], input_details["shape"]
+        )
+        tflite_interpreter.set_tensor(input_tensor_idx, input_data)
+
+      output_tensor_idx = tflite_interpreter.get_output_details()[0]["index"]
+      tflite_interpreter.invoke()
+      output_list.append(tflite_interpreter.get_tensor(output_tensor_idx))
+
+    tflite_interpreter.reset_all_variables()
+    for idx, input_details in enumerate(tflite_interpreter.get_input_details()):
+      input_tensor_idx = input_details["index"]
+      input_data = np.reshape(input_data_list[3][idx], input_details["shape"])
+      tflite_interpreter.set_tensor(input_tensor_idx, input_data)
+
+    output_tensor_idx = tflite_interpreter.get_output_details()[0]["index"]
+    tflite_interpreter.invoke()
+    output_list.append(tflite_interpreter.get_tensor(output_tensor_idx))
+    return output_list
+
+  def _use_tflm_interpreter(self, filename, input_data_list):
```

Comment:
Want to make these functions generic as possible other models/tests can use this. But if we want to make each test specific I can refactor these into the single test function for resource_variables. (Kind of became non-generic since we want to invoke a certain number of times and reset variables)